code/modules/projectiles/guncrafting/frame.dm

```diff
@@ -0,0 +1,77 @@
+/obj/item/weapon/gun/energy/frame
+	name = "energy gun frame"
+	desc = "A frame for constructing an Energy Gun."
+	customizable_type = CUSTOMIZABLE_ENERGY
+	icon = 'icons/obj/guncrafting/main.dmi'
+	icon_state = "frame_energy"
+	charge_sections = 2
+	ammo_type = list()
+
+/obj/item/weapon/gun/energy/frame/testing
+	pin = null
+
+/obj/item/weapon/gun_attachment/frame/energy
+	hacky_as_fuck = /obj/item/weapon/gun/energy/frame
```

Comment:
fuck you this was for some old debug shit

thanks for reminding me to remove it